src/OrleansRuntime/Hosting/IServiceProviderFactory.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime.Hosting
+{
+    /// <summary>
+    /// Abstraction for integrating <see cref="IServiceProviderFactory{TContainerBuilder}"/> implementations.
+    /// </summary>
+    internal interface IServiceProviderFactory
```

Comment:
I understand why this is here, but it might be a little bit confusing to have this `IServiceProviderFactory` be different than the generic interface with the same name in the DI library. Typically interfaces of the same name that just vary in their generic argument arity are very similar to each other.
Perhaps `IServiceProviderFactoryAdapter` or `IServiceProviderFactoryBuilder` are not necessarily more precise on their own, but at least different enough from the other to not confuse them with each other. Thoughts? I'm fine if we decide to keep it though.